### DISCUSSION TITLE
Is it possible to reuse "it" in nested test?

### DISCUSSION THREAD

**Discussion by iamyoki** [cypress-io/cypress#23716] [Questions and Help] [answered]
![image](https://user-images.githubusercontent.com/74389358/189054047-85e0316e-f5e6-4663-b703-a650f048be59.png)
![image](https://user-images.githubusercontent.com/74389358/189054064-3b6d1a0d-3f6c-4027-a28c-0912f5c76a25.png)

Say I have 3 second tabs under every primary tabs, and I have wrote those 2 tests:

```js
describe('Nested tabs', () => {
  it('swtich primary tabs', () => {
    primaryTabs.forEach((t) => {
      cy.contains(t).click();
    });
  });

  it('swtich second tabs', () => {
    secondTabs.forEach((t) => {
      cy.contains(t).click();
    });
  });
});
```

But that's not covering all clickable scenarios, what I want to achieve is like this:

```js
describe('Nested tabs', () => {
  it('swtich primary tabs', () => {
    primaryTabs.forEach((t) => {
      cy.contains(t).click();

      // Like this
      cy.reUseAnotherIt('swtich second tabs');
    });
  });

  it('swtich second tabs', () => {
    secondTabs.forEach((t) => {
      cy.contains(t).click();
    });
  });
});
```

Is there any better approach? The main goal is to make "it" nestable.

**Comment by emilyrohrbough** [comment, marked as answer]
It's not possible to re-use the `it` however you could pull out the logic into a re-usable function that is shared between tests or you could dynamically create these cases if it helped clean things up for you.
```js
describe('nested tabs', ()=> {
  ;[
      { name: 'primary', tabs: primaryTabs},
      { name: 'second', tabs: secondTabs},
   ].forEach(({ name, tabs }) => {
        it(`switches ${name} tabs`, () => {
             tabs.forEach((t) => {
                 cy.contains(t).click();
             })
         })
    })
})
```

**Reply by iamyoki** [reply]
Thank your. That's great!
